### PR TITLE
Feature/support abortcontroller

### PR DESCRIPTION
**Pull request by inject3r**
This commit introduces native support for AbortController in Axios core.

### Changes:
- Cancel request immediately if `config.signal.aborted` is true
- Attach `abort` event listener to abort ongoing requests via the configured adapter (e.g., xhr)
- Throw `CanceledError` when cancellation is triggered via AbortController
- Maintains full compatibility with the existing `cancelToken` implementation

This prepares the groundwork for broader Fetch-compatible cancellation patterns across adapters.

Adapters like `xhr.js` can optionally implement `.cancel()` logic using `xhr.abort()` to fully utilize this feature.

### CODE REVIEW

**Comment by WillianAgostini** [comment]
Hi @inject3r, thanks for your contribution! Could you please add tests for this feature? It helps make sure everything works as expected. Thanks!

**Comment by inject3r** [comment]
> Hi @inject3r, thanks for your contribution! Could you please add tests for this feature? It helps make sure everything works as expected. Thanks!

Hi, thanks for the feedback!
Absolutely, I'll work on adding tests for this feature and update the PR shortly.
Let me know if there are any specific edge cases or scenarios you'd like me to cover.

**Review by char23-web — APPROVED** [review]

File: lib/adapters/adapters.js [hunk 1]
```diff
@@ -8,31 +8,31 @@ const knownAdapters = {
...
 
 utils.forEach(knownAdapters, (fn, value) => {
   if (fn) {
     try {
-      Object.defineProperty(fn, 'name', {value});
-    } catch (e) {
-      // eslint-disable-next-line no-empty
+      Object.defineProperty(fn, 'name', { value });
+    } catch (_) {
+      // name property might be read-only; safely ignored
```
Comment (jasonsaayman) [inline lib/adapters/adapters.js:18]: please can you add back the eslint condition under this comment

File: lib/adapters/adapters.js [hunk 2]
```diff
@@ -42,7 +42,8 @@ export default {
       adapter = nameOrAdapter;
 
       if (!isResolvedHandle(nameOrAdapter)) {
-        adapter = knownAdapters[(id = String(nameOrAdapter)).toLowerCase()];
+        id = String(nameOrAdapter).toLowerCase();
+        adapter = knownAdapters[id];
```
Comment (jasonsaayman) [inline lib/adapters/adapters.js:46]: I don't really understand why we would want to make this change, it seems an unnecessary expansion of the same code with a temp variable

File: lib/adapters/adapters.js [hunk 3]
```diff
@@ -53,27 +54,31 @@ export default {
         break;
       }
 
-      rejectedReasons[id || '#' + i] = adapter;
+      rejectedReasons[id || `#${i}`] = adapter;
```
Comment (jasonsaayman) [inline lib/adapters/adapters.js:57]: nicely done

File: lib/adapters/adapters.js [hunk 4]
```diff
@@ -53,27 +54,31 @@ export default {
...
 
-      const reasons = Object.entries(rejectedReasons)
-        .map(([id, state]) => `adapter ${id} ` +
-          (state === false ? 'is not supported by the environment' : 'is not available in the build')
-        );
-
-      let s = length ?
-        (reasons.length > 1 ? 'since :\n' + reasons.map(renderReason).join('\n') : ' ' + renderReason(reasons[0])) :
-        'as no adapter specified';
+      const s = length
```
Comment (jasonsaayman) [inline lib/adapters/adapters.js:69]: if you are updating to use template literals here

**Review by jasonsaayman — COMMENTED** [review]
looks kind of good but i think we need to address the comments and we may need to consider adding tests

**Comment by inject3r** [comment]
> looks kind of good but i think we need to address the comments and we may need to consider adding tests

Yes, you're right about doing tests and writing documentation. I haven’t written any tests or documentation yet because I haven’t had much time, but I’ll make some time soon and add them.

**Comment by char23-web** [comment]
> > looks kind of good but i think we need to address the comments and we may need to consider adding tests
> 
> Yes, you're right about doing tests and writing documentation. I haven’t written any tests or documentation yet because I haven’t had much time, but I’ll make some time soon and add them.

**Comment by char23-web** [comment]
[Bun Tests](https://github.com/ChainSafe/lodestar/actions/workflows/test-bun.yml)

**Comment by jasonsaayman** [comment]
Thanks @inject3r. Unfortunately, we have already merged support for AbortController. I am going to close this PR.
